lib/src/ui/common/error/error_message_provider.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:allthenews/generated/l10n.dart';
+import 'package:allthenews/src/data/communication/api/api_exception.dart';
+import 'package:flutter/cupertino.dart';
+
+class ErrorMessageProvider {
+  static String provideErrorMessage(Exception exception, BuildContext context) =>
```

Comment:
czemu metody statyczne?

Reply:
zakładam, że będzie to używane w wielu miejscach apki, traktuje to jako mapowanie błędu do typu i nie widzę potrzeby tworzenie obiektu tej klasy i wygodniej będzie tego używać odnosząc się do instancji. Widzisz jakieś przeciwwskazania?

Reply:
oparłbym to o interfejs i jego implementację, wtedy daje to pełną elastyczność tj. możliwość wstrzykiwania, podmieniania implementacji czy mockowania w testach ;)

Reply:
a może extension na exceptionie?

Reply:
extension na pewno lepiej wygląda, ale wciąż nie daje żadnej elastyczności - niemniej jednak, jak nie podoba ci się opcja z interfejsem to dla mnie extension akceptowalny ;)